### PR TITLE
Adding set_auth_token() function

### PR DESCRIPTION
**Pull request by ThabetIbrahim**

### CODE REVIEW

**Comment by rkcudjoe** [comment]
@urkle, and these too.

**Comment by urkle** [comment]
Is there not a network request stubbing framework for PHP that you can use?

**Comment by ThabetIbrahim** [comment]
Well I'm using phpunit along with php-vcr library to record the http interactions

**Comment by urkle** [comment]
so, then why do the tests need to fetch the ENV for the app token / auth token?  those could be fixed values in the specs after you initially record.

**Comment by ThabetIbrahim** [comment]
Well they are used in all the test function to check the output received in the api whether it meets the expectations or not.

**Comment by urkle** [comment]
But they shouldn't need to if you are using a stub for network requests.   tests shouldn't have ENV vars like that.. as that means they could work for one person and not for another.

**Comment by ThabetIbrahim** [comment]
Oh ok i get it, i'll submit a new pull request tomorrow with the fixes.

**Comment by ThabetIbrahim** [comment]
One question if you don't mind so i wont misunderstand anything stubbing is providing a predefined answers to a function during a test to reduce the interactions with an outside server correct? but the client api depends heavily on the responds from the server so should i stub all the functions?

**Comment by urkle** [comment]
For any given test, you should stub any outside requests with prerecorded "expected" responses.  That way you do the following
- don't overburden an external service with requests for automated tests
- increase speed of those test (no need to contact external service)
- control the responses that come back.. 
  - this allows you to test your API library against invalid auth responses, etc.. in a controlled manner.

The objective of tests is to test your library, not tests the 3rd party service.  You are only testing that your library can parse and handle the responses you may receive from the 3rd party service. Thus you should stub any and all network requests that happen during a specific unit test.  Also the nature of a unit test is to test JUST the single unit.. (thus you stub everything outside of the unit)

**Comment by rkcudjoe** [comment]
@urkle do you mind reviewing these again?

File: README.md [hunk 1]
```diff
@@ -205,3 +205,14 @@ $hubstaff->screenshots("2016-05-22", "2016-05-24", array("projects"=>"112761"));
   ]
 }
 ```
+
+### Run tests
+
+In order to run user tests you need to get development dependencies using composer:
+```php
+composer install --dev
```
Comment (urkle) [inline README.md]: 👍 for using composer

**Comment by urkle** [comment]
So, did you not checkin the VCR yaml files?

**Comment by urkle** [comment]
otherwise this looks good once you check those in.

**Comment by ThabetIbrahim** [comment]
Awesome, well i have to do them while not subbing right? to record the http interactions. Sorry for the late response.

**Comment by urkle** [comment]
If the PHP VCR works like the Ruby one, if there is no "cassette" then it'll run the connections to the real internet and save/create the cassette (YML) files..  after that it'll only run off of the stubbed YML files.  You can also explicitly set it into record mode to re-record the yml files.

**Pull request merged**